### ISSUE TITLE
ESM module

### ISSUE THREAD

**Issue by talentlessguy** [paulmillr/readdirp#163]
Hello, thanks for making this module! I'm the author of [tinyhttp](https://tinyhttp.v1rtl.site) framework and it has `@tinyhttp/static` middleware which uses `readdirp`. As I noticed, it doesn't have a `"module"` field.

Is it a good idea to add this field to package.json so it will work? Or better not touch it?

**Comment by paulmillr** [comment]
Hi. It prob won’t work.

**Comment by talentlessguy** [comment]
@paulmillr hi, I made it work in other Node.js packages, here's one of them: https://github.com/talentlessguy/es-accepts

I can submit a pull request with adding the required fields and an ESM file

**Comment by paulmillr** [comment]
What is index.cjs, how does it look like?

What is the benefit of switching to cjs ?

**Comment by talentlessguy** [comment]
@paulmillr `.cjs` is basically the same as `.js` with `require`, but when you use modules, Node.is will recommend to use `.js` for ESM and `.cjs` for CommonJS

Another way is using `.mjs` for ESM and `.js` for CommonJS

**tldr;** `cjs` = `js` w/ `require`

**Comment by MVEMCJSUNPE** [comment]
Here are some relevant articles that explain what @talentlessguy is talking about:
[packages determining module system](https://nodejs.org/api/packages.html#packages_determining_module_system)
[hybrid npm packages](https://2ality.com/2019/10/hybrid-npm-packages.html)
[package checks](https://docs.skypack.dev/package-authors/package-checks#esm)

**Comment by calumk** [comment]
@talentlessguy - Is this fixed or no longer needed, or some other solution?

installing / running @tinyhttp/static provides the error:

`SyntaxError: Named export 'promise' not found. The requested module 'readdirp' is a CommonJS module, which may not support all module.exports as named exports.`

I know its  probably an error with @tinyhttp/static not with this repo, but since the discussion for it seems to be here, thought i would continue this thread.

**Comment by talentlessguy** [comment]
@calumk `@tinyhttp/static` has been deleted from the repo... probably I should also add a deprecation warning

use `sirv` instead

**Comment by paulmillr** [comment]
done
